lib/features/family/features/admin_fee/presentation/widgets/admin_fee_text.dart

```diff
@@ -18,25 +20,21 @@ class AdminFeeText extends StatefulWidget {
   final bool isMonthly;
   final bool isMultipleChildren;
 
-  @override
-  State<AdminFeeText> createState() => _AdminFeeTextState();
-}
-
-class _AdminFeeTextState extends State<AdminFeeText> {
-  final AdminFeeRepository _adminFeeRepository = getIt<AdminFeeRepository>();
-
   @override
   Widget build(BuildContext context) {
-    final theme = widget.theme ?? Theme.of(context);
-    final perchild = widget.isMultipleChildren ? ' per child' : '';
-    final fee =
-        _adminFeeRepository.getTotalFee(widget.amount).toStringAsFixed(2);
-    final monthlyText = 'Admin fee of \$$fee applies$perchild monthly';
-    final nonMonthlyText = 'An admin fee of \$$fee will be added';
-    return Text(
-      widget.isMonthly ? monthlyText : nonMonthlyText,
-      style: widget.textStyle ??
-          theme.textTheme.bodySmall!.copyWith(fontWeight: FontWeight.w700),
+    return BaseStateConsumer(
+      cubit: getIt<AdminFeeCubit>()..setAmount(amount),
```

Comment:
This is the only thing that's hacky/ i'm unhappy about. Normally I would define this as a field and pass it here. Then i would call init/ setAmount from initState/didChangeDependencies. The problem here is that I want this to be a "add and forget about it widget". When the widget this widget is nested in passes a different amount from the initial you can only know this new value via the build method.

2 solutions:
1) they way i've done it now
2) when the top widget changes the amount it should notify the AdminFeeCubit (that's the clean way to do it)

But it means that everywhere the AdminFee widget is being used you also need to notify the cubit and i just want this to be easy to use. Lemme know your thoughts!